src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
 Defines the 3 Scenic classes `Point`, `OrientedPoint`, and `Object`, and associated
 helper code (notably their base class `Constructible`, which implements the handling of
 property definitions and :ref:`specifier resolution`).
+
+.. warning::
+
+    In 2D compatibility mode, these classes are overwritten with 2D analogs. While
+    we make an effort to map imports to the correct class, this only works if imports
+    use the form ``import scenic.core.object_types as object_types`` followed by accessing
+    ``object_types.Object``. If you instead use ``from scenic.core.object_types import Object``,
+    you may get the wrong class.
+
 """
 
+import typing
+import warnings
```

Comment:
Since this PR touches almost every file anyway, let's take the opportunity to sort all the imports alphabetically (at least by package name -- don't worry about the specific names imported from each package), which makes it easier to see if a particular package has been imported. I've been trying to do that myself lately but there are probably still some old files where the order is arbitrary.

Reply:
I think we can use the `isort` tool for this, which we can hook in at the same time as `black`.